user-interface/src/lib/utils/semaphore.ts

```diff
@@ -0,0 +1,38 @@
+// TODO: Maybe this should be delegated to local storage... WHat happens if the map is scoped to unique tabs/windows if modules are not shared between tabs/windows?
```

Comment:
Is the concern that we would be retrieving the refresh token in one tab and breaking other tabs, thus causing a logout and forced login?

Reply:
Resource contention. Each window/tab has it's own semaphore so could, theoretically, leave the Okta refresh endpoint open to concurrent requests from different tabs. And yeah existing tokens are expired once a refresh is requested. Don't want to just hope the last refresh call to Okta is the one with the access token we store.